### DISCUSSION TITLE
Is there a Prisma eqivalent to Sequelize's findAndCountAll query?

### DISCUSSION THREAD

**Discussion by uncvrd** [prisma/prisma#2956] [Q&A] [answered]
Hi all! Title states my question. I am using Prisma for cursor based pagination. With Sequelize, I was able to use their [method](https://sequelize.org/master/class/lib/model.js~Model.html#static-method-findAndCountAll): `findAndCountAll()` to return my query along with the count of total records that would match this query so that I know when I hit the end of records to paginate from. 

Prisma has:
- `findMany()`
- `count()`

as two separate queries, at this point I guess I just need to query both and merge the results to get something similar or am I missing something on the `findMany()` method?

Thank you!

Here's how I'm currently using cursor based pagination. This works excellent except for the scenario where `args.first` and `results.length` are equal to each other in the final query because it'll send back a `has_next_page` of `true` event though there are exactly 0 records left to query. 

```
t.field('moreArtistRumors', {
  type: 'PaginatedRumors',
  args: {
    first: schema.intArg({ required: true }),
    cursor: schema.intArg(),
  },
  async resolve(_root, args, ctx) {

    const current_cursor = !args.cursor ? undefined : {
      id: args.cursor
    }

    const results = await ctx.db.rumor.findMany({
      take: args.first,
      skip: 1,
      cursor: current_cursor,
      orderBy: {
        id: "desc"
      }
    })

    const lastItem = results.length === args.first ? extractLastItem(results).id : null

    return {
      pagination_metadata: {
        cursor: lastItem,
        has_next_page: lastItem ? true : false // this works unless results.length === args.first on the final query (e.g. 40 === 40)
      },
      rumors: results
    }

  }
})
```

**Comment by ryands17** [comment, marked as answer]
Hey @uncvrd :wave: 
Currently there's no feature for that in Prisma, so could you create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) explaining your use case.

**Comment by SwapnilSoni1999** [comment]
I know I'm late but here's what I've tried 

```js
const [users, count] = await Promise.all([
            User.findMany({}),
            User.count(),
        ]);
        return { users, count };
```

in `findMany` you can pass your find query criteria and as well as in `count` as well

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
